### PR TITLE
feat: add huber loss

### PR DESCRIPTION
**Pull request by iProzd**

### CODE REVIEW

File: deepmd/utils/argcheck.py [hunk 1]
```diff
@@ -2446,6 +2446,14 @@ def loss_ener():
     doc_numb_generalized_coord = "The dimension of generalized coordinates. Required when generalized force loss is used."
     doc_relative_f = "If provided, relative force error will be used in the loss. The difference of force will be normalized by the magnitude of the force in the label with a shift given by `relative_f`, i.e. DF_i / ( || F || + relative_f ) with DF denoting the difference between prediction and label and || F || denoting the L2 norm of the label."
     doc_enable_atom_ener_coeff = "If true, the energy will be computed as \\sum_i c_i E_i. c_i should be provided by file atom_ener_coeff.npy in each data system, otherwise it's 1."
+    doc_use_huber = (
+        "Enables Huber loss calculation for energy/force/virial terms with user-defined threshold delta (D). "
+        "The loss function smoothly transitions between L2 and L1 loss: "
+        "- For absolute prediction errors within D: quadratic loss (0.5 * (error**2)) "
+        "- For absolute errors exceeding D: linear loss (D * |error| - 0.5 * D) "
+        "Formula: loss = 0.5 * (error**2) if |error| <= D else D * (|error| - 0.5 * D). "
+    )
```
Comment (njzjz) [inline deepmd/utils/argcheck.py:2455]: https://deepmodeling--4684.org.readthedocs.build/projects/deepmd/en/4684/train/train-input.html#argument:loss[ener]/use_huber

image

1. a linebreak may be needed before `-`. See [sel](https://deepmodeling--4684.org.readthedocs.build/projects/deepmd/en/4684/train/train-input.html#argument:model[standard]/descriptor[se_e3]/sel);
3. `|xxxx|` represents the link in the RST. May need to use `\|` (i.e. `"\\|"` in Python).

File: deepmd/pd/loss/ener.py [hunk 2]
```diff
@@ -88,6 +99,14 @@ def __init__(
...
         inference : bool
             If true, it will output all losses found in output, ignoring the pre-factors.
+        use_huber : bool
+            Enables Huber loss calculation for energy/force/virial terms with user-defined threshold delta (D).
+            The loss function smoothly transitions between L2 and L1 loss:
+            - For absolute prediction errors within D: quadratic loss (0.5 * (error**2))
+            - For absolute errors exceeding D: linear loss (D * |error| - 0.5 * D)
+            Formula: loss = 0.5 * (error**2) if |error| <= D else D * (|error| - 0.5 * D).
+        huber_delta : float
+            The threshold value used for Huber loss, controlling transition between L2 and L1 loss.
```
Comment (wanghan-iapcm) [inline deepmd/pd/loss/ener.py:109]: Is `huber_delta` the `D` in the formula provided with key `use_huber`? please clarify in the doc str.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]
